File: src/main/java/platform/ecommerce/dto/request/product/ProductCreateRequest.java
package platform.ecommerce.dto.request.product;

import jakarta.validation.constraints.*;
import lombok.Builder;

import java.math.BigDecimal;

/**
 * Product creation request DTO.
 */
@Builder
public record ProductCreateRequest(

        @NotBlank(message = "Product name is required")
        @Size(max = 200, message = "Product name cannot exceed 200 characters")
        String name,

        String description,

        @NotNull(message = "Base price is required")
        @DecimalMin(value = "0", message = "Base price cannot be negative")
        BigDecimal basePrice,

        Long categoryId
) {
}

File: src/main/kotlin/platform/ecommerce/dto/request/ProductCreateRequest.kt
package platform.ecommerce.dto.request

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive

data class ProductCreateRequest(
    @field:NotBlank(message = "SKU is required")
    val sku: String,
    @field:NotBlank(message = "Name is required")
    val name: String,
    val description: String,
    @field:Positive(message = "Price must be positive")
    val price: Double,
    @field:NotBlank(message = "Image URL is required")
    val imageUrl: String
)
